app/models/filtering_test.rb

```diff
@@ -12,6 +12,22 @@ def create_tasks
     save
   end
 
+  def archive_records
+    # If a filtering test for this product already has an archive, use it and copy the augmented records.
+    # Otherwise, create them.
+    first_filter_test = begin
+                          product.product_tests.filtering_tests.find_by(:patient_archive.ne => nil)
```

Comment:
This might be a place where the Ruby 2.3 `&.` (safe navigation) operator is cleaner than a `begin/rescue/end` block. See @rbclark's changes in https://github.com/projectcypress/cypress-validation-utility/blob/master/app/controllers/uploads_controller.rb#L48, but basically if it would throw a `NoMethodError` (e.g. the `find_by` returned `nil`), it just returns `nil` instead.

Reply:
`product&.product_tests&.filtering_tests&.find_by(:patient_archive.ne => nil)` for example

Reply:
The rescue in this case is actually due to the `find_by` throwing an exception. It is expected that all of the other parts will exist, there is just no way to get find_by to return nil for a single case, short of setting it to do so for the whole app.

Reply:
Ok, that's fine then. As much as I don't like the construct I see the reasoning behind it.